File: go-code/refactoring_api/preserve_whole_object/after/heating_plan.go
package after

import "log"

type TempRange struct {
	Low, High int
}

type Room struct {
	DaysTempRange TempRange
}

type Plan struct {
	TemperatureRange TempRange
}

//STEP 6: We dont need the old method aymore
/*func (plan Plan) withinRange(low, high int) bool {
	return (low >= plan.TemperatureRange.Low) && (high <= plan.TemperatureRange.High)
}*/

// STEP 1: create a new function withinRangeNew with TempRange signature
// STEP 5: Inline existing withinRange in new func
// STEP 7: Rename the new method to withinRange
func (plan Plan) withinRange(tempRange TempRange) bool {
	//STEP 2: call directly existing withinRange func
	//return plan.withinRange(tempRange.Low, tempRange.High)

	return (tempRange.Low >= plan.TemperatureRange.Low) && (tempRange.High <= plan.TemperatureRange.High)
}

func main() {
	aRoom := Room{DaysTempRange: TempRange{Low: 10, High: 25}}
	aPlan := Plan{} // Instantiate a Plan

	//STEP 4: Delete unused variables
	/*low := aRoom.DaysTempRange.Low
	high := aRoom.DaysTempRange.High*/

	//STEP 3: Call new method
	if !aPlan.withinRange(aRoom.DaysTempRange) {
		log.Fatalln("room temperature went outside range")
	}
	/*if !aPlan.withinRange(low, high) {
		log.Fatalln("room temperature went outside range")
	}*/
}

File: java-code/src/main/java/refactoring/refactoring_api/preserve_whole_object/after/HeatingPlan.java
package refactoring.refactoring_api.preserve_whole_object.after;

import java.util.ArrayList;
import java.util.List;

public class HeatingPlan {
    public static void main(String[] args) {
      Room aRoom = new Room(new TemperatureRange(10, 25));
      Plan aPlan = new Plan(); // Instantiate a Plan

      //STEP 4: Unused variables are not needed anymore
      /*int low = aRoom.getDaysTempRange().getLow();
      int high = aRoom.getDaysTempRange().getHigh();*/

      //STEP 3: Call the new method from the client
      /*if (!aPlan.withinRange(low, high)) {*/
      if (!aPlan.withinRange(aRoom.getDaysTempRange())) {
        List<String> alerts = new ArrayList<>();
        alerts.add("room temperature went outside range");
      }
    }
}
